x-pack/plugins/elastic_assistant/server/schemas/common.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import type * as rt from 'io-ts';
+import { exactCheck, formatErrors } from '@kbn/securitysolution-io-ts-utils';
+import type {
+  RouteValidationFunction,
+  RouteValidationResultFactory,
+  RouteValidationError,
+} from '@kbn/core/server';
+
+type RequestValidationResult<T> =
+  | {
+      value: T;
+      error?: undefined;
+    }
+  | {
+      value?: undefined;
+      error: RouteValidationError;
+    };
+
+export const buildRouteValidation =
```

Comment:
Note from review -- `io-ts` for now, but we can integrate `zod` along with an OpenAPI spec for generation going forward.